mysql-connector-cpp.yaml

```diff
@@ -0,0 +1,82 @@
+package:
+  name: mysql-connector-cpp
+  version: 9.0.0
+  epoch: 0
+  description: MySQL Connector/C++ is a MySQL database connector for C++
+  copyright:
+    - license: GPL-3.0-or-later
+
+environment:
+  contents:
+    packages:
+      - autoconf
+      - automake
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - cmake
+      - openssl-dev
+      - zlib-dev
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/mysql/mysql-connector-cpp
+      tag: ${{package.version}}
+      expected-commit: ac218f0d2d5b7957a8a2efee148bc8d7a1c623a5
+
+  - uses: cmake/configure
+
+  - uses: cmake/build
+
+  - uses: cmake/install
+
+  - working-directory: /home/build/output
+    runs: |
+      mkdir -p ${{targets.destdir}}/usr/lib
+      mv *.so* ${{targets.destdir}}/usr/lib/
+
+  - uses: strip
+
+subpackages:
+  - name: ${{package.name}}-dev
+    pipeline:
+      - uses: split/dev
+    dependencies:
+      replaces:
+        - ${{package.name}}
+    description: ${{package.name}} dev
+
+  - name: libmysqlclient
+    description: MySQL Connector/C++ shared libraries
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.contextdir}}/usr/lib
+          mv ${{targets.destdir}}/usr/lib/*.so* ${{targets.contextdir}}/usr/lib/
+          chmod 755 ${{targets.subpkgdir}}/usr/lib/*.so*
```

Comment:
After these splits. The main package contains incorrectly installed cmake files and INFO_BIN and INFO_SRC files. Imho the main package should just have the shared library.

The cmake files should be instructed to be correctly installed into -dev package. Ditto INFO_BIN and INFO_SRC files, not sure how they are used even, if at all.

And imho libmysqlclient as a sub-pckage should not exist at all.

Reply:
How should we proceed here? I was following the official build doc: https://dev.mysql.com/doc/connector-cpp/9.0/en/connector-cpp-installation-source-cpp.html#connector-cpp-installation-source-build

Do you think `libmysqlclient` package would redundant?

Reply:
the guide only lists libraries, it doesn't tell you the name of the subpackage to use, and how one splits them or not.

So yes, I do believe the libmysqlclient subpackage should be re-dundant, and the default files in destdir should be cleaned up. (correct cmake files locations, and likely purge the weird INFO files if they are unreferenced otherwiese).

Reply:
Thanks @xnox, I have removed the sub-package. But I'm not sure if I understand the "INFO infos", could you please take a look when possible?

Reply:
Kindly ping here. 🙃 @xnox